### PR TITLE
Qw ii rz cleanup form interacted with

### PR DESCRIPTION
**Pull request by roblevintennis**
# For now pointed to qwII-rz-cleanup but after that's merged we'll point this to master

___

**Description:**
- [Asana Task](https://app.asana.com/0/1137984670709817/1142882503761970/f)
- Form interacted with is now exposed to consumer

**Referencing PR or Branch (e.g. in CMS or monorepo):**
- 

**Screenshots:**
image

### CODE REVIEW

File: src/components/Inputs/ButtonSelectGroup/ButtonSelectGroup.md [hunk 1]
```diff
@@ -33,7 +33,7 @@
 ```jsx
 <ButtonSelectGroup
   defaultValue="excellent"
-  label="Health"
+  labelCopy="Health"
```
Comment (roblevintennis) [inline src/components/Inputs/ButtonSelectGroup/ButtonSelectGroup.md]: This was that button select group error we kept seeing

File: src/hooks/useFormState.js [hunk 2]
```diff
@@ -29,6 +29,10 @@ export function useFormState(initialState) {
     }
   }
 
+  function getFormInteractedWith() {
+    return touched
```
Comment (roblevintennis) [inline src/hooks/useFormState.js]: I double checked `setStateFactory` which is where this is set and basically the first time a button select group is clicked, or, a character is entered into a field (text, birthdate, zip) `touched` (for the form) becomes true. So, this works for our purposes.

I was thinking of exposing `getFieldValues` but Nick pointed out that was meant to be a private variable, and also I was thinking if we somehow checked all field values are `null` to determine if the form was interacted with, that could get pretty inefficient with a form that has a lot of fields.

Happy that we already had a flag to check and expose this.

File: src/components/Form/Form.js [hunk 3]
```diff
@@ -116,7 +117,7 @@ export function Form({ children, config }) {
           fieldConfig.validators
             .reduce((errors, validator) => errors.concat(validator(field)), [])
             .filter((x) => !!x) // remove empty strings
-            .join(', '),
+            .join('. '),
```
Comment (roblevintennis) [inline src/components/Form/Form.js]: I've  removed the double error messaging  we saw on Zip, but, in the event we do have multiple, these should be sentances e.g. "Please do this. Please do that."

File: src/components/Form/Form.js [hunk 4]
```diff
@@ -138,7 +139,12 @@ export function Form({ children, config }) {
     <form onSubmit={onSubmit}>
       {/* See the top of this file or ./Form.md for help on how to use
         these arguments passed to the children function. */}
-      {children(field, getFormErrorMessage, getFormIsValid)}
+      {children(
+        field,
+        getFormErrorMessage,
+        getFormIsValid,
+        getFormInteractedWith
```
Comment (roblevintennis) [inline src/components/Form/Form.js]: @iamvictorli this will unblock you

File: src/components/Form/Form.md [hunk 5]
```diff
@@ -39,24 +38,31 @@ import { TitleLarge, TextInput, Spacer, Button, InfoMessage, ZipInput, ZipInputV
     },
     onSubmit: async (formData) => {
       await new Promise((resolve) => setTimeout(resolve, 500))
-      if (!!(Math.floor(Math.random() * 10) % 2)) {
-        throw new Error("Oh no, the api is broken (try again, it's random)")
+      if (count++ % 2 === 0) {
```
Comment (roblevintennis) [inline src/components/Form/Form.md]: Multiple requests to make this  deterministic (sorry @kenning  😛 cc / @anbright @ekeric13 I believe y'all provided this feedback)

File: src/components/Form/Form.md [hunk 6]
```diff
@@ -39,24 +38,31 @@ import { TitleLarge, TextInput, Spacer, Button, InfoMessage, ZipInput, ZipInputV
...
   }}
 >
-  {(field, getFormErrorMessage, getFormIsValid) => (
+  {(field, getFormErrorMessage, getFormIsValid, getFormInteractedWith) => (
     <div>
       <TitleLarge.Serif.Book500>Example Form</TitleLarge.Serif.Book500>
 
       <Spacer.H16 />
 
+      {!!getFormInteractedWith() && (
```
Comment (roblevintennis) [inline src/components/Form/Form.md]: @iamvictorli just an example usage, but you will easily be able to call:
```js
      {!!getFormInteractedWith() && ( removeCTAButton() }
```

File: src/components/Form/Form.js (hunk 4 above)
Comment (iamvictorli) [reply src/components/Form/Form.js]: 👍

File: src/components/Form/Form.md (hunk 6 above)
Comment (roblevintennis) [reply src/components/Form/Form.md]: and you probably don't need the `!!` part

Comment (iamvictorli) [reply src/components/Form/Form.md]: @roblevintennis You probably dont need the `!!` operator, since the getFormInteractedWith already returns a boolean

**Comment by roblevintennis** [comment]
I'm going  to merge this as it's pointed to another  topic branch anyway. Also, I won't to have it's base branch mergable soon, and so I'll need to point CMS to the latest here.

**Pull request merged**